aten/src/ATen/native/native_functions.yaml

```diff
@@ -7737,12 +7737,20 @@
 # TODO: remove dispatch section when porting TH CUDA to ATen
 - func: multinomial.out(Tensor self, int num_samples, bool replacement=False, *, Generator? generator=None, Tensor(a!) out) -> Tensor(a!)
   dispatch:
-    CPU, CUDA: multinomial_out
+    CPU, CUDA, MPS: multinomial_out
+
+- func: multinomial_with_replacement_mps(Tensor self, int num_samples, *, Generator? generator=None) -> Tensor
```

Comment:
These independent registrations in the native_functions is not a good idea. We should remove these and add separate function even if there is copy of the code.

Reply:
Done